src/file.ts

```diff
@@ -514,6 +515,43 @@ export class File {
     return fsp.unlink(this.pathname);
   }
 
+  /**
+   * Recursively delete the folder and contents.
+   *
+   * @memberOf File
+   * @method
+   * deleteRecursively
+   * @return void
+   * @example
+   * import File from 'file-js';
+   *
+   * const file = new File('./dir/');
+   * file.deleteRecursively();
+   */
+  public async deleteRecursively(dirPath: string = this.pathname): Promise<void> {
+    if (existsSync(dirPath)) {
+      const files = readdirSync(dirPath);
```

Comment:
Any reason for using `sync` calls? Can could use:
- `this.exists`
- `getList` (or `getFiles`) 
- `this.isDirectory`

Reply:
`this.exists` can be used.  

`getList`/`getFiles` returns the full filepath as a string so will be harder to extract the filenames, but can be done. `readDirSync` seemed like a cleaner solution.

couldn't get `this.isDirectory` to work